app/jobs/application_job.rb

```diff
@@ -0,0 +1,2 @@
+class ApplicationJob < ActiveJob::Base
```

Comment:
Since we have `ApplicationJob` now, maybe it would make sense to merge `BaseWorker` into this class. While we're at it we could also replace the weird `process -> process_delegate` setup with the more conventional `ActiveJob::Base#perform` and an AJ callback